### PR TITLE
Dbt refactor

### PR DESCRIPTION
**Pull request by ieaves**

### CODE REVIEW

**Comment by edlouth** [comment]
Should we be including pycache files in source control?

File: image/src/integrations/flat_file/__init__.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+args = Args()
+assert os.path.exists(args.file), f"{args.file} does not exist"
+
+
+def get_nodes_and_edges(client):
+    nodes, edges = base.get_nodes_and_edges(args.file, args.namespace)
+    nodes = adapt_to_client(nodes)
+    # edges = adapt_to_client(edges)
+    return nodes, []
```
Comment (edlouth) [inline image/src/integrations/flat_file/__init__.py:22]: What happened to edges here?

File: image/src/integrations/dbt/__init__.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import os
+from dataclasses import dataclass
+
+from grai_source_dbt import base
+
+
+@dataclass
+class Args:
+    manifest_file: str = os.environ["GRAI_DBT_MANIFEST_FILE"]
+    namespace: str = os.environ.get("GRAI_NAMESPACE", "default")
```
Comment (edlouth) [inline image/src/integrations/dbt/__init__.py:10]: Do we need to default here if the action has a default?  I note we haven't got a default in image/src/main.py

File: action.yml [hunk 3]
```diff
@@ -23,7 +23,7 @@ inputs:
   workspace:
```
Comment (edlouth) [inline action.yml:23]: Should this file be deleted?

**Review by edlouth — CHANGES_REQUESTED** [review]

File: dbt/action.yaml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    description: 'The port for the Grai server'
+    required: false
+    default: "443"
+  api-key:
+    description: 'A user api key for server access'
+    required: true
+  grai-frontend-host:
+    description: "The url for your frontend, e.g. http://localhost:3000"
+    required: false
+    default: app.grai.io
```
[the remark this answers is not in the document]
Comment (ieaves) [reply dbt/action.yaml:30]: It's not necessary. The client will attach https for port 443 or http for port 80 if they aren't provided.

File: image/src/integrations/flat_file/__init__.py (hunk 1 above)
Comment (ieaves) [reply image/src/integrations/flat_file/__init__.py:22]: Flat file doesn't currently produce edges =/.

File: image/src/integrations/dbt/__init__.py (hunk 2 above)
Comment (ieaves) [reply image/src/integrations/dbt/__init__.py:10]: Need is a strong word - I don't think it's needed for the code to function (though it doesn't introduce any bugs either). The reason I included a default here was a signal to future viewers that the value was optional and to treat it as such. I don't

File: image/entrypoints/flat-file.sh [hunk 5]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+pip install grai-source-flat-file
```
[the remark this answers is not in the document]
Comment (ieaves) [reply image/entrypoints/flat-file.sh:3]: The python3-slim variants have python3 under the base python path so this doesn't actually break anything but good to be explicit in any case!

**Comment by ieaves** [comment]
> Should we be including pycache files in source control?

No, looks like this repo didn't get initialized with an appropriate gitignore. Should be fixed now.

**Review by edlouth — APPROVED** [review]

**Pull request merged**